barretenberg/cpp/src/barretenberg/ultra_honk/protogalaxy.test.cpp

```diff
@@ -1,345 +1,412 @@
-        alpha = FF::random_element();
+    static std::shared_ptr<Instance> fold_and_verify(const std::vector<std::shared_ptr<Instance>>& instances,
+                                                     Composer& composer,
+                                                     bool expected_result)
+    {
+        auto folding_prover = composer.create_folding_prover(instances);
+        auto folding_verifier = composer.create_folding_verifier();
+
+        auto proof = folding_prover.fold_instances();
+        auto next_accumulator = proof.accumulator;
+        auto res = folding_verifier.verify_folding_proof(proof.folding_data);
+        EXPECT_EQ(res, expected_result);
+        return next_accumulator;
     }
-    auto full_honk_evals = ProtoGalaxyProver::compute_full_honk_evaluations(
-        instance->prover_polynomials, instance->alphas, instance->relation_parameters);
 
-    // Evaluations should be 0 for valid circuit
-    for (const auto& eval : full_honk_evals) {
-        EXPECT_EQ(eval, FF(0));
+    static void check_accumulator_target_sum_manual(std::shared_ptr<Instance>& accumulator, bool expected_result)
+    {
+        auto instance_size = accumulator->instance_size;
+        auto expected_honk_evals = ProtoGalaxyProver::compute_full_honk_evaluations(
+            accumulator->prover_polynomials, accumulator->alphas, accumulator->relation_parameters);
+        // Construct pow(\vec{betas*}) as in the paper
+        auto expected_pows = PowPolynomial(accumulator->gate_challenges);
+        expected_pows.compute_values();
+
+        // Compute the corresponding target sum and create a dummy accumulator
+        auto expected_target_sum = FF(0);
+        for (size_t i = 0; i < instance_size; i++) {
+            expected_target_sum += expected_honk_evals[i] * expected_pows[i];
+        }
+        info("expected_target_sum", expected_target_sum);
+
+        EXPECT_EQ(accumulator->target_sum == expected_target_sum, expected_result);
     }
-}
-TEST_F(ProtoGalaxyTests, PerturbatorCoefficients)
-{
-    std::vector<FF> betas = { FF(5), FF(8), FF(11) };
-    std::vector<FF> deltas = { FF(2), FF(4), FF(8) };
-    std::vector<FF> full_honk_evaluations = { FF(1), FF(1), FF(1), FF(1), FF(1), FF(1), FF(1), FF(1) };
-    auto perturbator = ProtoGalaxyProver::construct_perturbator_coefficients(betas, deltas, full_honk_evaluations);
-    std::vector<FF> expected_values = { FF(648), FF(936), FF(432), FF(64) };
-    EXPECT_EQ(perturbator.size(), 4); // log(instance_size) + 1
-    for (size_t i = 0; i < perturbator.size(); i++) {
-        EXPECT_EQ(perturbator[i], expected_values[i]);
+    static void decide_and_verify(std::shared_ptr<Instance>& accumulator, Composer& composer, bool expected_result)
+    {
+        auto decider_prover = composer.create_decider_prover(accumulator);
+        auto decider_verifier = composer.create_decider_verifier(accumulator);
+        auto decision = decider_prover.construct_proof();
```

Comment:
I'm nit picking now but the name `decision` here doesn't make sense to me. It's basically a Honk proof right? (With an error term in sumcheck yadda yadda). I can see calling the output of the  `decider_verifier` the `decision` if you like